roles/kubernetes-apps/rotate_tokens/tasks/main.yml

```diff
@@ -3,6 +3,9 @@
   shell: "{{ bin_dir }}/kubectl get secrets -o custom-columns=name:{.metadata.name} --no-headers | grep -m1 default-token"
   register: default_token
   changed_when: false
+  until: default_token.rc == 0
```

Comment:
can you please explain why this needed? why kubectl call can failed on this step?

Reply:
As explained in the commit message, in some installation, it can take up to 5sec to get the value. Retrying
for 15 sec will ensure the command won't return 1 and fail for nothing.

Reply:
oh, sorry, i miss commit message 😞 
but i don't understand why `kubectl get secret` can not reachable in this task. 
Think this fix hides a symptom

Reply:
No worries. Hum I don't think this is really concerning, it's more a timing issue that can be introduced by deploying in VMs IMHO.

Reply:
This role running after network_plugins role and it can be network issue.
What network plugin you using? I'm checking `Vagrant ` file changes with calico plugin, all's good.

Reply:
I'm using the default networking plugin, I've changed that part. But once again that's probably a timing issue with the virtual machines. It's hard to diagnose and a bit hard to predict. The play will pass in 2/3 seconds and the timeout is 15sec which is acceptable IMHO. I'm sure other people running in virtual environments will encounter this issue.
Thanks.

Reply:
you mean running kubespray playbooks manually on VM`s without vagrant?
can you please provide your inventory, i'm will trying to reproduce this error

Reply:
Running on Vagrant + libvirt.

Here's my `config.rb`:

```
$os = "centos"
$box = "centos/7"
$etcd_instances = 1
$kube_master_instances = 1
$num_instances = 9
$kube_node_instances_with_disks = true
$kube_node_instances_with_disks_size = "20G"
$kube_node_instances_with_disks_number = 2
```

Do you have a vagrant-libvirt setup?

Reply:
not yet, need some time for running it

Reply:
this ok for reproduce?
```
➜  kubespray cat vagrant/config.rb
$os = "centos"
$box = "centos/7"
$etcd_instances = 1
$kube_master_instances = 1
$num_instances = 9
$kube_node_instances_with_disks = true
$kube_node_instances_with_disks_size = "20G"
$kube_node_instances_with_disks_number = 2
➜  kubespray head -n 8 roles/kubernetes-apps/rotate_tokens/tasks/main.yml
---
- name: Rotate Tokens | Get default token name
  shell: "{{ bin_dir }}/kubectl get secrets -o custom-columns=name:{.metadata.name} --no-headers | grep -m1 default-token"
  register: default_token
  changed_when: false
  # until: default_token.rc == 0
  # delay: 5
  # retries: 3
➜  kubespray vagrant up --provider=libvirt
DEPRECATION: The 'sudo' option for the Ansible provisioner is deprecated.
Please use the 'become' option instead.
The 'sudo' option will be removed in a future release of Vagrant.

Bringing machine 'k8s-01' up with 'libvirt' provider...
Bringing machine 'k8s-02' up with 'libvirt' provider...
Bringing machine 'k8s-03' up with 'libvirt' provider...
Bringing machine 'k8s-04' up with 'libvirt' provider...
Bringing machine 'k8s-05' up with 'libvirt' provider...
Bringing machine 'k8s-06' up with 'libvirt' provider...
Bringing machine 'k8s-07' up with 'libvirt' provider...
Bringing machine 'k8s-08' up with 'libvirt' provider...
Bringing machine 'k8s-09' up with 'libvirt' provider...
==> k8s-08: Box 'centos/7' could not be found. Attempting to find and install...
    k8s-08: Box Provider: libvirt
    k8s-08: Box Version: >= 0
==> k8s-08: Loading metadata for box 'centos/7'
    k8s-08: URL: https://vagrantcloud.com/centos/7
==> k8s-08: Adding box 'centos/7' (v1801.02) for provider: libvirt
    k8s-08: Downloading: https://vagrantcloud.com/centos/boxes/7/versions/1801.02/providers/libvirt.box
...
```

Reply:
Should be ok, yes.

Reply:
can't reproduce, all 5 runs finished successfully 😞

Reply:
can you please provide:

 - os
 - ansible version
 - vagrant version
 - `vagrant up` output for failed run

Reply:
now with your `config.rb`  inventory looks like

```
[etcd]
k8s-0[1:1]

[kube-master]
k8s-0[1:1]

[kube-node]
k8s-0[1:3]
```

i'm changing `$num_instances = 9` to `$kube_node_instances=9` and trying again

Reply:
I have this:

```
# Generated by Vagrant

k8s-07 ansible_host=192.168.121.220 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.107 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-02 ansible_host=192.168.121.58 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.102 bootstrap_os=centos local_release_dir=/vagrant/temp downlo
ad_run_once=False kube_network_plugin=flannel
k8s-01 ansible_host=192.168.121.232 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.101 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-04 ansible_host=192.168.121.157 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.104 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-05 ansible_host=192.168.121.231 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.105 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-06 ansible_host=192.168.121.135 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.106 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-09 ansible_host=192.168.121.228 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.109 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-08 ansible_host=192.168.121.233 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.108 bootstrap_os=centos local_release_dir=/vagrant/temp downl
oad_run_once=False kube_network_plugin=flannel
k8s-03 ansible_host=192.168.121.65 ansible_port=22 ansible_user='vagrant' ansible_ssh_private_key_file='/home/leseb/.vagrant.d/insecure_private_key' ip=172.17.8.103 bootstrap_os=centos local_release_dir=/vagrant/temp downlo
ad_run_once=False kube_network_plugin=flannel

[etcd]
k8s-0[1:1]

[kube-master]
k8s-0[1:1]

[kube-node]
k8s-0[1:9]

[k8s-cluster:children]
kube-master
kube-node
```

Reply:
Sorry, but i cant reproduce bug, trying 5 times with 5 nodes.

You can open issue with more info about error for understand what wrong with apiserver access after install network plugin. 

Remove please changes from `rotate_tokens/tasks/main.yml` and i can merge changes in `Vagrantfile`.
Thanks!

Reply:
Thanks, I'll remove and open a separate issue.